fix: drop --path and similar from bundler invocations

pierrewillenbrockdfki opened the pull request:
Of note here is the different meaning of --path between bundle install and bundle binstubs, where the former uses the configuration entry "path", while the later uses "bin", which they are saying in their documentation but are not clear that those refer to the configuration variables.

doudou committed:
chore: ignore Gemfile.lock

doudou committed:
fix: teardowm of cli/test_watch.rb if the setup failed in the middle

doudou committed:
fix: generate stub XML files that are not empty

Newer rexml versions started rejecting empty files as invalid XML

pierrewillenbrockdfki committed:
fix: drop --path and similar from bundler invocations

Of note here is the different meaning of --path between bundle install
and bundle binstubs, where the former uses the configuration entry "path",
while the later uses "bin", which they are saying in their documentation
but are not clear that those refer to the configuration variables.

chhtz commented on lib/autoproj/package_managers/bundler_manager.rb:
```diff
@@ -303,8 +303,11 @@ def self.run_bundler_install(
+                run_bundler(ws, "config", "set", "--local", "shebang", Gem.ruby,
+                            bundler_version: bundler_version)
+
                 options << "--binstubs" << binstubs if binstubs
```
Just removing that option works for me. Not sure if there are scenarios where this is necessary, or if stubs are installed by `install_autoproj(...)` anyways.

pierrewillenbrockdfki committed:
fix: do not pass --binstubs to bundle install

instead, set the configuration option before and
call bundle binstubs --all afterwards.

doudou commented:
Just extracted the "fix CI" commits from the other PR #426
